apps/messaging-api/src/routes/message-actions/index.ts

```
@@ -0,0 +1,144 @@
+      if (messageOptionId !== req.body.messageId) {
+        throw new BadRequestError(
+          errorProcess,
+          "url params id mismatch with body id",
+        );
+      }
+
+      const userId = ensureUserIdIsSet(req, errorProcess);
+
+      /**
+       * We can opt to have link tables for each option, or have a table with dynamic option type/value
+       *
+       * This is easier to change than the public contract
+       *
+       * Right now, the only option is "is seen"
+       */
+      let updateQueryResult:
+        | QueryResult<{
+            isSeen: boolean;
+          }>
+        | undefined;
+
+      const jobEvents: { type: EventType; data: MessageEventData }[] = [];
+      let poolClient: PoolClient | undefined;
+      try {
+        poolClient = await app.pg.pool.connect();
+        console.log("broder tuck");
+        console.log(userId, req.body.messageId);
+        const existanceCheckQueryResult = await poolClient.query<{
+          exists: boolean;
+        }>(
+          `
+            select exists(
+              select * from messages m 
+              join users u on (u.user_profile_id = $1 or u.id::text = $1)
+              where (u.id::text = $1 or u.user_profile_id = $1) and m.id = $2
+              limit 1
+            )
+          `,
+          [userId, req.body.messageId],
+        );
+
+        if (!existanceCheckQueryResult.rows.at(0)?.exists) {
+          throw new NotFoundError(errorProcess, "message not found for user");
+        }
+
+        // seen/unseen
+        updateQueryResult = await poolClient.query<{ isSeen: boolean }>(
+          `
+            with message_selection as(
+                select m.id from messages m
+                join users u on (u.user_profile_id = $1 or u.id::text = $1)
+                where 
+                    (u.id::text = $1 or u.user_profile_id = $1)
+                    and is_delivered = true 
+                    and m.id = $2
+                    and $3 != m.is_seen
```

Do we want to return an error if an already seen message is set as seen again?

---

No with motivation:

(big) if we add more actions, we'll treat this as an ordinary put. There's nothing inherently criminal with putting a resource to the same. Just nothing happens. I think this is normal behaviour.

---

Sorry, I misread the code totally fine